packages/aws-cdk/test/api/hotswap/s3-bucket-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,863 @@
+
+    // THEN
+    expect(deployStackResult).toBeUndefined();
+    expect(mockLambdaInvoke).not.toHaveBeenCalled();
+  });
+
+  test('calls the lambdaInvoke() API when it receives an asset difference in two s3 bucket deployments and IAM Policy differences using old-style synthesis', async () => {
+    // GIVEN
+    let s3Deployment2 = {
+      Type: 'Custom::CDKBucketDeployment',
+      Properties: {
+        ServiceToken: {
+          'Fn::GetAtt': [
+            'S3DeploymentLambda',
+            'Arn',
+          ],
+        },
+        SourceBucketNames: [
+          'src-bucket-old',
+        ],
+        SourceObjectKeys: [
+          'src-key-old',
+        ],
+        DestinationBucketName: 'DifferentBucketOld',
+      },
+    };
+
+    setup.setCurrentCfnStackTemplate({
+      Resources: {
+        ServiceRole: serviceRole,
+        Policy1: policy,
+        Policy2: policy2,
+        S3DeploymentLambda: deploymentLambda,
+        S3Deployment: s3Deployment,
+        S3Deployment2: s3Deployment2,
+      },
+    });
+
+    policy.Properties.PolicyDocument.Statement[0].Resource = {
+      'Fn::GetAtt': [
+        'WebsiteBucketNew',
+        'Arn',
+      ],
+    };
+
+    policy2.Properties.PolicyDocument.Statement[0].Resource = {
+      'Fn::GetAtt': [
+        'DifferentBucketNew',
+        'Arn',
+      ],
+    };
+
+    s3Deployment.Properties.SourceBucketNames = ['src-bucket-new'];
+    s3Deployment.Properties.SourceObjectKeys = ['src-key-new'];
+    s3Deployment.Properties.DestinationBucketName = 'WebsiteBucketNew';
+
+    s3Deployment2.Properties.SourceBucketNames = ['src-bucket-new'];
+    s3Deployment2.Properties.SourceObjectKeys = ['src-key-new'];
+    s3Deployment2.Properties.DestinationBucketName = 'DifferentBucketNew';
+
+    const cdkStackArtifact = setup.cdkStackArtifactOf({
+      template: {
+        Resources: {
+          ServiceRole: serviceRole,
+          Policy1: policy,
+          Policy2: policy2,
+          S3DeploymentLambda: deploymentLambda,
+          S3Deployment: s3Deployment,
+          S3Deployment2: s3Deployment2,
+        },
+      },
+    });
+
+    // WHEN
+    const deployStackResult = await cfnMockProvider.tryHotswapDeployment(cdkStackArtifact);
+
+    // THEN
+    expect(deployStackResult).not.toBeUndefined();
+
+    payload.ResourceProperties = {
+      SourceBucketNames: [
+        'src-bucket-new',
+      ],
+      SourceObjectKeys: [
+        'src-key-new',
+      ],
+      DestinationBucketName: 'WebsiteBucketNew',
+    };
+
+    expect(mockLambdaInvoke).toHaveBeenCalledWith({
+      FunctionName: 'arn:aws:lambda:here:123456789012:function:my-deployment-lambda',
+      Payload: JSON.stringify(payload),
+    });
+
+    payload.ResourceProperties.DestinationBucketName = 'DifferentBucketNew';;
+
+    expect(mockLambdaInvoke).toHaveBeenCalledWith({
+      FunctionName: 'arn:aws:lambda:here:123456789012:function:my-deployment-lambda',
+      Payload: JSON.stringify(payload),
+    });
+  });
+});
```

Comment:
I have to say, I find these `old-style synthesis` tests very difficult to follow. Perhaps you can shorten them a little bit, introduce some types, or do the mutations inline...? It's really hard to see what these are testing right now.